File: Deep Sequencing/Data/gsean.pl
#Find overlapping regions of subsets of opposite transcripters with genes. read subset.perl. write final/curated.out
#Find overlapping asRNA regions with psbA1, read full.perl, write stdout

use Text::CSV;
use Bio::SeqFeature::Generic;
use Bio::FeatureIO;


my $fin = Bio::FeatureIO->new(-file=>'elongatus.ptt', -format=>'ptt');
my $psb;
my @genes;
while (my $f = $fin->next_feature) {
  push @genes, $f;
  if (($f->get_tag_values('Synonym'))[0] eq 'Synpcc7942_0424') {
    $psb = $f;
  }
}
print $psb->location->start;
print "\n";

my @transcripts;
my $csv = Text::CSV->new();
open (CSV, "<", 'full.perl') or die $!;
while (<CSV>) {
  $csv->parse($_);
  my @r = $csv->fields();
  if ($r[4] eq '+') {
    $strand=1;
  }
  else {
    $strand=-1;
  }
  $feat = new Bio::SeqFeature::Generic ( -start => $r[1], -end => $r[0],
                                -strand => $strand,
                                -score  => 1000,
                                -tag => {
                                  chrome=>$r[3],
                                  pearson=>$r[5]
                                }
                                     );
  push @transcripts, $feat;
  #~ #antisense RNA
  if ($feat->overlaps($psb) and ($feat->strand == -1) and (($feat->get_tag_values('pearson'))[0] lt 0)) {
    print $feat->location->start;
    print " ";
    print $feat->location->end;
    print " ";
    print $feat->strand;
    print " ";
    @a=$feat->get_tag_values('pearson');
    print $a[0];
    print "\n";
  }
  #~ foreach (@genes) {
    #~ if ($feat->overlaps($_) and ($feat->strand == $_->strand) and (($feat->get_tag_values('chrome'))[0] eq 'CP000100')) {
      #~ print $feat->location->start;
      #~ print " ";
      #~ print $feat->location->end;
      #~ print " ";
      #~ print $feat->strand;
      #~ print " ";
      #~ @a=$feat->get_tag_values('pearson');
      #~ print $a[0];
      #~ print "\n";
      #~ #COG, Length, PID
      #~ $pro = ($_->get_tag_values('Product'))[0];
      #~ $id = ($_->get_tag_values('Synonym'))[0];
      #~ print "$id\n$pro\n\n";
    #~ }
  #~ }
} 
close CSV;

File: Deep Sequencing/Data/gsean.py
#Create a subset of opposite transcripters with psbA1, use data by gse.py(total.mat)
#Make a map from id to start stop positions, write to subset/full.perl
from scipy.io import loadmat
import csv
p=loadmat('total.mat')['p']
l=p[0:,1].copy()
l.sort()
#print l

p[:,[1,0]]=p[:,[0,1]]
ps = dict(p)

f = open("GPL13535-11452.txt")
while f.readline()[0]=='#':
  pass
val = list()
rows = csv.reader(f, delimiter='\t')
genes = 0
for r in rows:
  val.append(dict())
  val[genes]['chr']=r[1]
  val[genes]['db']=r[2]
  val[genes]['strand']=r[3]
  val[genes]['start']=r[4]
  val[genes]['stop']=r[5]
  genes += 1
print genes
f.close()

#stop start chr db strand
f = open('full.perl','w')
w = csv.writer(f)
for i in l:
  v = val[int(i-1)]
  v['pearson'] = ps[i]
  #print val[int(i-1)]
  w.writerow([v['stop'],v['start'], v['chr'], v['db'], v['strand'], v['pearson']])
f.close()
